ast/src/asm_analysis/display.rs

```diff
@@ -40,7 +40,7 @@ impl<T: Display> Display for Machine<T> {
         write_items_indented(f, &self.registers)?;
         write_items_indented(f, &self.instructions)?;
         write_items_indented(f, &self.callable)?;
-        write_items_indented(f, &self.constraints)?;
+        write_items_indented(f, &self.pil)?;
```

Comment:
Is it a problem that these might be ordered differently? I mean can we always re-parse stuff that is printed here?

Reply:
Can you elaborate here? Within constraints, the order is preserved throughout the compiler, but eventually this should not be a requirement, right? There are no tests which parse outputs of this display afaik, this could make sense to add but it seems unrelated to this change.

Reply:
I'm just wondering if it might make sense to move the pil a bit further up or what happens if you mix constraints and instruction declarations and such, but yes, unrelated to this PR.